### ISSUE TITLE
Import aliases for Qual

### ISSUE THREAD

**Issue by bruth** [dave/jennifer#32]
Hi, thanks for the great package. I had two questions regarding the generated import aliases.

I noticed that doing something like `Qual("context", "Context")` would explicitly set the import alias, e.g. `import context "context"` even when it doesn't conflict with other imports. Is this by design to explicitly set the alias for all imports?

The second question is being able to set the default import alias explicitly (in lieu of naming conflict resolution). For example, I have a package each having a sub-package that is named after the version: `github.com/foo/bar/v1`, `github.com/foo/bar/v2`. The `package` names for both are set to `bar` since they are the `bar` package but just for different versions of the API. However when using `Qual` the import alias is set to `v1` and `v2` rather than the actual package name.

Since this *is* generated code, I realize this should not matter, however some of the generated code I am producing are stubs that will be further edited by humans, so this is why I am concerned with aesthetics. Thanks!

**Comment by bruth** [comment]
Unrelated, but I just read the "use cases" respond on the Wiki.. I think the other advantage to using this library is compiler-time checks. I hate having to trace down what issue in my input data structure is causing a template to blow up.

**Comment by bruth** [comment]
Funny enough I just found [this commit](https://github.com/MrEhbr/jennifer/commit/39bc3d015ff32cf470f85d00e485404001c83734) that of a fork seems to address my questions..

**Comment by dave** [comment]
This has been asked a few times, and I was [sent a PR](https://github.com/dave/jennifer/pull/31) last week that suggested a solution. 

I've put the emphasis on code that works rather than formatting and aesthetics, so it's something I've resisted. 

However, if this is something that's continually requested I'll have a think about supporting it. If I was to implement this, it probably wouldn't be in the same way as the #31 suggested...

Something like:

```go
f := jen.NewFile("main")
f.AddPackageName("github.com/dave/foo", "foo") // package foo will use default name "foo"
f.AddPackageAlias("github.com/dave/bar", "baz") // package bar will be aliased to "baz"
f.Func().Id("main").Params().Block(
    jen.Qual("github.com/dave/foo", "Foo").Call(),
    jen.Qual("github.com/dave/bar", "Bar").Call()
)
// Output:
// package main
// 
// import (
//     "github.com/dave/foo"
//     baz "github.com/dave/bar"
// )
// 
// func main() {
//     foo.Foo()
//     baz.Bar()
// }
```

**Comment by dave** [comment]
I'll continue thinking about this for a few more days.

**Comment by dave** [comment]
... and perhaps I could add package name defaults for all the standard library packages, so you wouldn't need: `f.AddImportName("fmt", "fmt")`

**Comment by bruth** [comment]
Thanks for the consideration Dave and I understand the focus on simplicity. I think your solution of being explicit and not changing anything existing is a good solution. If a concern is dealing with correct package names, then using these functions could have a "use at your own risk" disclaimer noting the [edge cases](https://github.com/MrEhbr/jennifer/pull/1#commitcomment-23639584) one could run into.

**Comment by sharonjl** [comment]
I am not fully aware of all the features of `Qual` but something like this might work too,
```
QualAs(path, name, alias string)
```

**Comment by dave** [comment]
I'm not keen on adding this feature to the `Qual` method... It raises questions:

* Do you have to specify the alias on all the `Qual` methods? 
* Or just one? 
* Does it have to be the first one?
* What happens if two `QualAs` methods are used with different aliases?

I'd much prefer to make the link between package path and alias name once, and leave `Qual` just specifying the package path as it does currently.

**Comment by bruth** [comment]
> I'd much prefer to make the link between package path and alias name once, and leave Qual just specifying the package path as it does currently.

👍 I agree.

**Comment by dave** [comment]
OK take a look in here: https://github.com/dave/jennifer/tree/33 - let me know if you can see any edge cases that will break it. I'll continue testing tomorrow.

**Comment by dave** [comment]
OK I've merged the PR and tagged it v0.18.0. Let me know if you have any problems... and thanks!

**Comment by bruth** [comment]
Wonderful, thank you! It is working great so far.
